node/node_test.go

```diff
@@ -0,0 +1,83 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	proxy "github.com/cometbft/cometbft/proxy"
+	cmtypes "github.com/cometbft/cometbft/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rollkit/rollkit/config"
+	test "github.com/rollkit/rollkit/test/log"
+	"github.com/rollkit/rollkit/types"
+)
+
+// cleanUpNode stops the node and checks if it is running
+func cleanUpNode(node Node, t *testing.T) {
+	defer func() {
+		if fn, ok := node.(*FullNode); ok {
+			fn.cancel()
+		}
+		if ln, ok := node.(*LightNode); ok {
```

Comment:
I don't think a node can be both a `FullNode` and a `LightNode`, so isn't it better to use an `else if` here?